### PR TITLE
~3x speedup for b62 decoding

### PR DESCRIPTION
**Pull request by maxtaco**
@oconnor663 / @patrickxb this was the big speedup I found in saltpack tonight

### CODE REVIEW

File: go/encoding/basex/stream.go [hunk 1]
```diff
@@ -242,7 +242,7 @@ func newDecoder(enc *Encoding, r io.Reader) io.Reader {
 	return &decoder{
 		enc:        enc,
 		r:          r,
-		buf:        make([]byte, 128*enc.base256BlockLen),
-		scratchbuf: make([]byte, 128*enc.baseXBlockLen),
+		buf:        make([]byte, 8192*enc.base256BlockLen),
+		scratchbuf: make([]byte, 8192*enc.baseXBlockLen),
```
Comment (oconnor663) [inline go/encoding/basex/stream.go]: Are we sure this part helps?

**Comment by oconnor663** [comment]
LGTM, with the one question.

**Comment by maxtaco** [comment]
Thought of another idea for a speedup.  Instead of multiplying by 62 every
time, what about multiplying by 64 (bitshift left by 6) followed by
subtracting a multiplication by 2 (bitshift left by 1)?

Would have to micro benchmark it. It could in fact be slower if it is
expensive to copy intermediate results.

File: go/encoding/basex/stream.go (hunk 1 above)
Comment (maxtaco) [inline go/encoding/basex/stream.go]: It did help about ~5% and it will help more in the RPC-proxying case.

**Pull request merged**